### PR TITLE
magento/magento2-login-as-customer#144: "Login as Customer" functionality should be enabled by default.

### PR DESCRIPTION
**Pull request by engcom-Foxtrot**
### Fixed Issues (if relevant)

1. Fixes magento/magento2-login-as-customer#144: "Login as Customer" functionality should be enabled by default.
2. Fixes magento/magento2-login-as-customer#175: Fix performance degradation

### Manual testing scenarios (*)

1. See magento/magento2-login-as-customer#144: "Login as Customer" functionality should be enabled by default.
2. Also this PR can affect any other "Login as Customer" functionality.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
**Result:**
"Login as Customer" functionality is enabled by default.
image

**_Was checked cases with Disable Page Cache For Admin User=Yes/No:_**

- Admin logout finished customer session
adminLogout


- Basic "Login as Customer" functionality: place order, add product to wish list/compare list, edit customer address/account information, persistent banner is shown on every page
PlaceOrder

File: app/code/Magento/LoginAsCustomerApi/Api/SetLoggedAsCustomerAdminIdInterface.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\LoginAsCustomerApi\Api;
+
+/**
+ * Set id of Admin logged as Customer.
+ *
+ * @api
```
Comment (naydav) [inline app/code/Magento/LoginAsCustomerApi/Api/SetLoggedAsCustomerAdminIdInterface.php:13]: Need to remove `@api` annotation

File: app/code/Magento/LoginAsCustomerFrontendUi/Plugin/KeepLoginAsCustomerSessionDataPlugin.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
```
Comment (naydav) [inline app/code/Magento/LoginAsCustomerFrontendUi/Plugin/KeepLoginAsCustomerSessionDataPlugin.php:6]: please add declare strict type (check all files)

Comment (engcom-Foxtrot) [reply app/code/Magento/LoginAsCustomerFrontendUi/Plugin/KeepLoginAsCustomerSessionDataPlugin.php:6]: @naydav done.

File: app/code/Magento/LoginAsCustomerApi/Api/SetLoggedAsCustomerAdminIdInterface.php (hunk 1 above)
Comment (engcom-Foxtrot) [reply app/code/Magento/LoginAsCustomerApi/Api/SetLoggedAsCustomerAdminIdInterface.php:13]: @naydav done.

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
Was checked cases from [comment](https://github.com/magento/magento2-login-as-customer/pull/182#issuecomment-648853973)

**Comment by engcom-Foxtrot** [comment]
Closed due to https://github.com/magento/magento2-login-as-customer/pull/194.

**Pull request closed without merging**